test/powershell/engine/Remoting/RemoteSession.Basic.Tests.ps1

```diff
@@ -10,47 +9,266 @@ Describe "New-PSSession basic test" -Tag @("CI") {
+            $closedSession.Runspace.Close()
+            $openSession = New-RemoteSession -ConfigurationName $endPoint
+
+            $ParameterError = @(
+                @{
+                    parameters    = @{
+                        'InDisconnectedSession' = $true
+                        'AsJob'                 = $true
+                        'ScriptBlock'           = {1}
+                        'ComputerName'          = 'localhost'
+                        'ConfigurationName'     = $endpoint
+                    }
+                    expectedError = 'System.InvalidOperationException,Microsoft.PowerShell.Commands.InvokeCommandCommand'
+                    title         = 'Cannot use InDisconnectedState and AsJob together'
+                },
+                @{
+                    parameters    = @{
+                        'ScriptBlock' = {1}
+                        'SessionName' = 'SomeSessionName'
+                    }
+                    expectedError = 'System.InvalidOperationException,Microsoft.PowerShell.Commands.InvokeCommandCommand'
+                    title         = 'Cannot use SessionName without InDisconnectedSession'
+                },
+                @{
+                    parameters    = @{
+                        'ScriptBlock' = { 1 }
+                        'Session'     = $disconnectedSession
+                        'ErrorAction' = 'Stop'
+                    }
+                    expectedError = 'InvokeCommandCommandInvalidSessionState,Microsoft.PowerShell.Commands.InvokeCommandCommand'
+                    title         = 'Cannot use Invoke-Command on a disconnected session'
+                }
+                @{
+                    parameters    = @{
+                        'ScriptBlock' = { 1 }
+                        'Session'     = $closedSession
+                        'ErrorAction' = 'Stop'
+                    }
+                    expectedError = 'InvokeCommandCommandInvalidSessionState,Microsoft.PowerShell.Commands.InvokeCommandCommand'
+                    title         = 'Cannot use Invoke-Command on a closed session'
+                }
+            )
+
+            function script:ValidateSessionInfo($session, $state)
+            {
+                $session.ComputerName | Should BeExactly 'localhost'
+                $session.ConfigurationName | Should BeExactly $endPoint
+                $session.State | Should Be $state
+            }
+        }
+    }
+
+    AfterAll {
+        $global:PSDefaultParameterValues = $originalDefaultParameterValues
+
+        if($isWindows)
+        {
+            Remove-PSSession $disconnectedSession,$closedSession,$openSession -ErrorAction SilentlyContinue
+        }
+    }
+
+    It 'Can connect to default endpoint' {
+        $session = New-RemoteSession -ConfigurationName $endPoint
+        ValidateSessionInfo -session $session -state 'Opened'
+        $session | Remove-PSSession -ErrorAction SilentlyContinue
```

Comment:
This session won't get cleaned up during test failure. Comment also applies to the other It's within this Context.

Reply:
Fixed. Used try/finally